src/tinuous/github.py

```diff
@@ -143,6 +145,15 @@ def get_event_id(self, run: WorkflowRun, event_type: EventType) -> str:
                         )
                     except IndexError:
                         pr = "UNK"
+                    except RateLimitExceededException:
+                        log.warning(
+                            "Search rate limit exceeded; not performing any"
+                            " more issue searches"
+                        )
+                        pr = "UNK"
+                        self.doing_searches = False
+                else:
+                    pr = "UNK"
```

Comment:
that would make runs "non-deterministic" and vary depending on how many `search` requests are still left.  I think implementing waiting (https://github.com/con/tinuous/issues/51) at this point and retrying would be a better solution even though at a cost of the `tinuous` run potentially taking long time, but at least it should then "do its best" and "consistently" produce the same results.